libraries/AP_SensorHead/examples/Master/Master.cpp

```diff
@@ -0,0 +1,138 @@
+#include <AP_SensorHead/AP_SensorHead.h>
+#include <AP_SensorHead/AP_SensorHead_Stream.h>
+
+#include <AP_BoardConfig/AP_BoardConfig.h>
+#include <AP_Scheduler/AP_Scheduler.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/utility/RingBuffer.h>
+#include <SITL/SITL.h>
+
+const AP_HAL::HAL &hal = AP_HAL::get_HAL();
+using namespace SensorHead;
+
+class UnknownMessageHandler : public AP_SensorHead_Handler<UnknownMessage> {
+public:
+    void handle(UnknownMessage::data_t *data)
+    {
+        hal.console->printf("Handling Unknown\n");
+    }
+};
+
+
+// TODO: Should this eventually be a "vehicle"?
+class SensorHead_Master : public AP_HAL::HAL::Callbacks {
+public:
+    // HAL::Callbacks implementation.
+    void setup() override;
+    void loop() override;
+    void receive();
+
+    SensorHead_Master(AP_HAL::Stream *in, AP_HAL::Stream *out) :
+        shead(AP_SensorHead::init()),
+        inputStream(in),
+        outputStream(out),
+        param_loader(var_info)
+    {}
+
+private:
+    static const AP_Scheduler::Task scheduler_tasks[];
+    static const AP_Param::Info var_info[];
+
+    SITL::SITL sitl;
+
+    AP_Scheduler scheduler;
+
+    AP_SensorHead *shead;
+    UnknownMessageHandler unknownHandler;
+
+    AP_Baro baro;
+    AP_InertialSensor ins;
+
+    // NOTE: Separate streams to be able to loopback using named pipes.
+    AP_HAL::Stream *inputStream;
+    AP_HAL::Stream *outputStream;
+    AP_SensorHead_Stream shead_stream{shead, inputStream, outputStream};
+
+    AP_Param param_loader;
+
+    void ins_update(void);
+    void baro_update(void);
+    void baro_accumulate(void);
+};
+
+
+static SensorHead_Master master{hal.uartC, hal.uartD};
+
+#define SCHED_TASK(func, _interval_ticks, _max_time_micros) SCHED_TASK_CLASS(SensorHead_Master, &master, func, _interval_ticks, _max_time_micros)
+
+const AP_Scheduler::Task SensorHead_Master::scheduler_tasks[] = {
+    SCHED_TASK(ins_update,             50,   1000),
```

Comment:
I don't think using AP_Scheduler is ideal here. For this example I think we should be running the CPU flat out, polling all sensors in turn.